test/staking/TokenStaking.test.js

```diff
@@ -2027,6 +2102,148 @@ describe("TokenStaking", () => {
     )
   })
 
+  describe("forceDecreaseAuthorization", () => {
+    context("when application is not approved", () => {
+      it("should revert", async () => {
+        await expect(
+          tokenStaking
+            .connect(auxiliaryAccount)
+            .forceDecreaseAuthorization(
+              operator.address,
+              application1Mock.address
+            )
+        ).to.be.revertedWith("Application is not disabled")
+      })
+    })
+
+    context("when application is approved", () => {
+      it("should revert", async () => {
+        await tokenStaking
+          .connect(deployer)
+          .approveApplication(application1Mock.address)
+        await expect(
+          tokenStaking
+            .connect(deployer)
+            .forceDecreaseAuthorization(
+              operator.address,
+              application1Mock.address
+            )
+        ).to.be.revertedWith("Application is not disabled")
+      })
+    })
+
+    context("when application is paused", () => {
+      it("should revert", async () => {
+        await tokenStaking
+          .connect(deployer)
+          .approveApplication(application1Mock.address)
+        await tokenStaking
+          .connect(deployer)
+          .setPanicButton(application1Mock.address, panicButton.address)
+        await tokenStaking
+          .connect(panicButton)
+          .pauseApplication(application1Mock.address)
+        await expect(
+          tokenStaking
+            .connect(staker)
+            .forceDecreaseAuthorization(
+              operator.address,
+              application1Mock.address
+            )
+        ).to.be.revertedWith("Application is not disabled")
+      })
+    })
+
+    context("when application is not authorized", () => {
```

Comment:
Should we wrap and the next context into another one saying "when application is disabled"? Something like:
```
context("when application is disabled", () => {
  context("when application was not authorized", () => {
  }

  context("when application was authorized", () => {
  }
})
```

Reply:
My suggestion was based on the fact I was confused this application is disabled *and* not authorized. I was about to write a comment it is not possible to disable unapproved application (I missed approval with authorization). 

I think `context` is not only about extracting common setup but also about... wrapping elements into some additional information context. But if we want to avoid it, maybe we can provide more info in test names:
- when application was not authorized and god disabled
- when application was authorized and got disabled
? 

Both options are equally good for me.

Reply:
good point, renamed
✔️